### ISSUE TITLE
[Epic] Networking

### ISSUE THREAD

**Issue by nelsonic** [dwyl/home#133]
## Context

At present we are using the default Wifi Access Point / Router provided by the ISP #107. 
This is fine while the house is still "in development", but undesirable for a number of reasons the first being a lack Wifi coverage and the last a lack of security (threat monitoring/detection). Also, the ISP (_and every person in tech support who works for them_) has full access to our internal network including all our devices and traffic. 🙄 
We need to put a decent firewall between the ISP Router and our internal network and then _disable_ the Wifi on the ISP Router. 

We have _already_ purchased a Ubiquti Networks Unifi **16-port POE Gigabit Switch** https://www.ui.com/unifi-switching/unifi-switch-16-150w and a Pack of 5 Unifi Long Range **Access Points** https://www.ui.com/unifi/unifi-ap-ac-lr  
So we have the Wifi covered (_in terms of equipment_) for both indoor and exterior internet access.
There are **28 RJ45 cables** connecting the basement to the rest of the house. One RJ45 Port in each room and several in corridors. The ceilings of the Garage, Leisure Room and Co-working/Classroom all have ports on the Ceiling in order to connect projectors or other equipment. 

If we chose to use all the RJ45 Ports, which I suspect we will, we will need to purchase another 16/24 Port Switch. (TBD)
The _reason_ I have not purchased the rest of the equipment yet, is because we need to do a full assessment of our _needs_. e.g: do we _need_ all the RJ45 Ports to be POE or will a regular (_much cheaper_) switch suffice?
I have a strong preference for using Ubiquti hardware https://www.ui.com because they have _superb_ UX and security. It's _always_ more expensive than the _cheapest_ (TPLink or Huawei), but this is an area I don't think we want to skimp on.

# Todo

+ [ ] Build/Buy Networking Switch Cabinet/Rack to store/protect all our networking equipment.
Needs to fit the equipment we end up buying (see below).
+ [ ] Research and buy remaining network equipment

# Research

+ [ ] UniFi Dream Machine Pro: https://store.ui.com/collections/unifi-network-routing-switching/products/udm-pro
+ [x] Unifi Security Gateway https://www.ui.com/unifi-routing/usg Firewall and Network control system
+ [ ] External Security Cameras #130.

I have been putting off setting up the network because I expect it to take several days and is only relevant when we _need_ advanced networking features such as the POE for the Security System / Door Entry. Help very much appreciated.

**Comment by iteles** [comment]
We've been speaking a bit more about this with the first step being going through all the cables, identifying which one goes to each room and adding the 'plug ends' to them so that they're usable within the switch.

I spoke to our electricians about this to see if it was part of their original remit, which it wasn't, but they have provided the following quote:

## Materials

![image](https://user-images.githubusercontent.com/4185328/99988603-a96abe00-2da9-11eb-9f83-5f4d8c9c3e0e.png)

This is the brand of cabinet and equipment they are proposing, **Barpa**:
https://barpa.eu/products/network-cabinet-norma-2/

You can see the measurements of the proposed cabinet in the quote: 600 x 600 x 1670
![image](https://user-images.githubusercontent.com/4185328/99988868-ea62d280-2da9-11eb-9ff8-fb135c503b8d.png)


## Labour
This would be for going through all the cables, identifying each one, adding the plastic 'plugs' on the end, installing the cabinet/rack on the wall and adding all the cables to it.

**Estimate:** 2 days x 2 men, totalling **€384 + IVA**.

Note: This is an estimate, if more time is required, they will charge us for more time, but they're usually pretty spot on with their labour estimates.

## Timelines
They will have to order the equipment but predict they would be able to get this done **next week** if we were to go with this cabinet/rack. If we can get our _own_ cabinet/rack by next week then they can also keep to that schedule.

The advantage here also being that when they're here, we can hopefully get them to stay and finish the *snag list* #108 as well.

**Comment by nelsonic** [comment]
There are ISPs with smaller racks than that! Are they having a _laugh_? 🤷🏼‍♂️

**Comment by iteles** [comment]
No, I think they have no idea what we're after 🤷‍♀️  They think we're 'very techy' 😆 
Do they think we need to fit the hole mass of cables in it?

This is the smallest size of _these_ cabinets they do:
![image](https://user-images.githubusercontent.com/4185328/100018035-b6040c00-2dd3-11eb-85d3-82ca779066ab.png)

But if you just send me a link to something akin to what we're actually after, that will make it a lot faster to explain.

**Comment by nelsonic** [comment]
The smallest size this particular site sells is actually this MiniLan box:
https://barpa.eu/products/wall-cabinet-minilan-crux-3
![image](https://user-images.githubusercontent.com/194400/100019891-bbaf2100-2dd6-11eb-9905-f08eb4e8b24f.png)

Sadly, it's too narrow for the UniFi 16-port POE Switch (_that we already own_).
https://store.ui.com/collections/unifi-network-routing-switching/products/usw-16-poe
![image](https://user-images.githubusercontent.com/194400/100020218-5871be80-2dd7-11eb-9556-5d5efc65df1c.png)

This one is bigger: https://barpa.eu/products/wall-cabinet-crux-1/
![image](https://user-images.githubusercontent.com/194400/100019734-7985df80-2dd6-11eb-82b2-e12e1538184e.png)

![image](https://user-images.githubusercontent.com/194400/100019768-89052880-2dd6-11eb-9106-bd4d20d79072.png)

![image](https://user-images.githubusercontent.com/194400/100019799-95898100-2dd6-11eb-8528-e58e5d57adf1.png)

@iteles I was thinking of getting a **4U Cabinet** e.g:
https://www.ebay.co.uk/itm/4U-Server-Rack-Network-Cabinet-19-inch-600-x-450mm-Black/274364712671

![image](https://user-images.githubusercontent.com/194400/100028336-9b875e00-2de6-11eb-8fe3-a88330b89e39.png)

We _really_ don't need more than that. We aren't going to run any actual servers (beyond a couple of Raspberry Pis),
so the _most_ we are going to store in the rack is _two_ of the POE Switches mentioned above. 
Which means we probably only need a 2U rack, but those tend to be _more_ expensive and difficult to find. 🙄 

This is the most reasonably priced 4U rack I found in PT:
https://cablematic.com/pt/produtos/armario-de-rack-19-4u-mural-600x450x286mm-gama-sohorack-asw-diy-WY001/
![image](https://user-images.githubusercontent.com/194400/100030294-16eb0e80-2deb-11eb-97a5-445690e1f196.png)

**Comment by nelsonic** [comment]
@iteles can you please explain to the electricians that we have "modest" networking needs. 

Even with 5 POE Security Cameras, 5 POE Access Points and the electronic doors using Tom's setup,
we only need two 16-port POE Switches and a **4U Rack/Cabinet** is _ample_ for our needs. 

Please ask them if they have a supplier for a similar rack to the one I shared above, or if we should order it ourselves?

I'm happy to mount the rack where I want it ahead of their arrival or to advise them where I think it should go. 
Also happy for them to help us with the installation, but I honestly don't understand why it will take _two_ people two days. 🤷‍♂️ 
But given that it would probably take me/us more than 2 days (_like a whole weekend_) to do it, I'm happy to pay them.
Basically we will ask them to wire up _all_ the cables in the house and then we can "hack" the ones we decide to use for the electronic doors later. 

Thanks. ❤️

**Comment by iteles** [comment]
Thanks for the research and for clarifying this! It did seem a bit large but I didn't know if you had discussed requirements with them yet.

I've emailed them and hope they come back with something soon!

**Comment by nelsonic** [comment]
Finished setting up the Unifi Security Gateway: https://github.com/dwyl/learn-computer-networking/issues/6 ✅
